### DISCUSSION TITLE
Upgrade from 1.7.9 (sqlite) to 1.8.x fails

### DISCUSSION THREAD

**Discussion by 0xphk** [YOURLS/YOURLS#3275] [Plugins]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/YOURLS/.github/blob/main/CODE_OF_CONDUCT.md).

### Submission validity

- [X] This is not a personal support request, that should be posted on [discussions community](https://github.com/YOURLS/YOURLS/discussions).
- [X] I checked [current issues](https://github.com/YOURLS/YOURLS/issues) and this request isn't a duplicate of an existing issue, opened or closed.

### Self troubleshooting

- [X] I've read the [documentation](https://yourls.org/) and made sure to follow it.
- [X] I've read the [troubleshooting first steps](https://github.com/YOURLS/YOURLS/wiki/Troubleshooting-first-steps) guide.

### Version

1.7.9

### Description

Working installation is 1.7.9 using sqlite as database.

After overwriting current installation with newer files from 1.8.2, I only get error 500 with following stack trace in nginx error log:

```
2022/04/06 19:29:25 [error] 450814#450814: *9176 FastCGI sent in stderr: 
"PHP message: PHP Notice:  Undefined variable: dbhost in /var/www/yourls/user/db.php on line 46 
PHP message: PHP Fatal error:  Uncaught Error: Call to undefined function yourls_get_db() in /var/www/yourls/includes/functions-debug.php:20
Stack trace:
#0 /var/www/yourls/user/db.php(46): yourls_debug_log()
#1 /var/www/yourls/user/db.php(9): yourls_db_sqlite_connect()
#2 /var/www/yourls/includes/Config/Init.php(148): require_once('/var/www/yourls...')
#3 /var/www/yourls/includes/Config/Init.php(57): YOURLS\Config\Init->include_db_files()
#4 /var/www/yourls/includes/load-yourls.php(26): YOURLS\Config\Init->__construct()
#5 /var/www/yourls/admin/index.php(3): require_once('/var/www/yourls...')
#6 {main}
thrown in /var/www/yourls/includes/functions-debug.php on line 20" while reading response header from upstream, client: x.x.x.x,
server: xxx.xxx, request: "GET /admin/ HTTP/2.0", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "xxx.xxx"
```

### Context

Current 1.7.9 installation was installed from git clone.

Sqlite db is simply a file ```yourls.sq3``` in the user directory.

Tested with files from 1.8.0 too but with the same result. 

```define( 'YOURLS_DEBUG', true );``` was set but produced the same stack trace output in nginx log as above.

Any help appreciated.

**Comment by dgw** [comment]
I have migrated this to a discussion as it's not a bug in YOURLS.

Core does not have SQLite support, which is why you have a `user/db.php` file. That's where the error comes from, according to the traceback. Are you using the (quite old) SQLite driver plugin from [here](https://github.com/ozh/yourls-sqlite)?

**Reply by 0xphk** [reply]
I guess this could be the source, was already wondering why the file was missing in the archive. 
The file looks a little bit different but the header says its for v1.7.3 and from Ozh.

```
/var/www/yourls/user/db.php                                                                                                                                                                                                                            
<?php
/*
SQLite driver for YOURLS.
Version: 1.1
This driver requires YOURLS 1.7.3 -- not before -- not after!
Author: Ozh
*/
```

**Reply by dgw** [reply]
You'll need an updated version of the DB layer. Try https://github.com/Flameborn/yourls-sqlite

**Reply by 0xphk** [reply]
Thanks for the link to the updated version.

**Reply by 0xphk** [reply]
Small update, the interface now loads after using the above mentioned version of yourls-sqlite with v1.8.2, 
but the upgrade itself wasn't successful. 
After running ```Upgrade``` in the interface, the following errors came up:
```
Unable to update the DB.
You will have to manually fix things, sorry for the inconvenience :(
The errors were:
SQLSTATE[HY000]: General error: 1 near "DATABASE": syntax error
SQLSTATE[HY000]: General error: 1 near "CONVERT": syntax error
SQLSTATE[HY000]: General error: 1 near "CHANGE": syntax error
SQLSTATE[HY000]: General error: 1 near "CHANGE": syntax error
SQLSTATE[HY000]: General error: 1 near "CHANGE": syntax error
SQLSTATE[HY000]: General error: 1 near "CONVERT": syntax error
```
Afterwards no webinterface was showing up with 'Service unavailable' status 503.

But setting up a new sqlite db using the ```Install``` option on a fresh extracted v1.8.2 instead, the command worked and a new working db was setup.

As there were only a few static shortened URLs, I added them manually and now only the stats are gone which isn't a big deal.

So thanks again, my version is now up2date.

**Reply by ozh** [reply]
Yeah, I didn't cover the upgrade procedure in my initial plugin, so I guess the fork that @dgw mentionned doesn't either

**Comment by 0xphk** [comment]
Marked as solved, setting up a new sqlite db using the updated db layer worked.
